### PR TITLE
refactor: versioning.is functions return boolean

### LINKED ISSUE #4873: Refactor versioning is* functions to return boolean
Returning string has potential for confusion. Probably requires creation of a "massage" function.

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Refactor versioning.is* functions to return only `boolean`

## Context:

Closes #4873 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/versioning/composer/index.ts [hunk 1]
```diff
@@ -102,17 +102,17 @@ const isGreaterThan = (a: string, b: string): boolean =>
 const isLessThanRange = (version: string, range: string): boolean =>
   npm.isLessThanRange(composer2npm(version), composer2npm(range));
 
-const isSingleVersion = (input: string): string | boolean =>
-  input && npm.isSingleVersion(composer2npm(input));
+const isSingleVersion = (input: string): boolean =>
+  !!(input && npm.isSingleVersion(composer2npm(input)));
```
Comment (viceice) [inline lib/versioning/composer/index.ts:106]:
```suggestion
  !!input && npm.isSingleVersion(composer2npm(input));
```

Should do it

File: lib/versioning/loose/generic.ts [hunk 2]
```diff
@@ -19,12 +19,11 @@ export interface VersionComparator {
 // parse should return null if version not valid
 // parse should return an object with property release, an array of version sections major.minor.patch
 export const parser = (parse: VersionParser): Partial<VersioningApi> => {
-  function isValid(version: string): string {
```
Comment (viceice) [inline lib/versioning/loose/generic.ts:22]: this code is refactored in #13481, so ignore and revert

File: lib/versioning/ubuntu/index.ts [hunk 3]
```diff
@@ -11,26 +11,25 @@ const temporarilyUnstable = ['22.04'];
...
-  _range?: string
-): string | boolean | null {
+function isCompatible(version: string, _range?: string): boolean {
   return isValid(version);
 }
 
-function isSingleVersion(version: string): string | boolean | null {
-  return isValid(version) ? true : null;
+function isSingleVersion(version: string): boolean {
+  return isValid(version) ? true : false;
```
Comment (viceice) [inline lib/versioning/ubuntu/index.ts:32]:
```suggestion
  return isValid(version);
```

simplify

**Comment by rarkins** [comment]
@viceice enforcing range in `isCompatible()` resulted in more refactoring than I was comfortable with. I suggest to leave it as-is

**Comment by viceice** [comment]
> @viceice enforcing range in `isCompatible()` resulted in more refactoring than I was comfortable with. I suggest to leave it as-is

OK, but we should then do it in separate PR.

**Review by viceice — APPROVED** [review]

**Pull request merged**